tests/codeception/acceptance/LocalFooterCest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Class LocalFooterCest.
+ *
+ * @group local_footer
+ */
+class LocalFooterCest {
+
+  /**
+   * Only site manager and higher should have access.
+   */
+  public function testAccess(AcceptanceTester $I) {
+    $I->logInWithRole('contributor');
+    $I->amOnPage('/admin/appearance/local-footer');
+    $I->canSeeResponseCodeIs(403);
+    $I->amOnPage('/user/logout');
+    $I->logInWithRole('site_manager');
+    $I->amOnPage('/admin/appearance/local-footer');
+    $I->canSeeResponseCodeIs(200);
+  }
+
+  /**
+   * Changes to the local footer should display correctly.
+   */
+  public function testCustomLocalFooter(AcceptanceTester $I) {
+    $I->logInWithRole('site_manager');
+    $I->amOnPage('/admin/appearance/local-footer');
+    $I->checkOption('Enabled');
+    $I->selectOption('State', 'New York');
+    $fields = [
+      'Company' => 'Drupal',
+      'Street address' => '123 Drupal Dr',
+      'City' => 'New York',
+      'Zip code' => 12345,
+      'su_local_foot_action[0][uri]' => 'http://google.com',
+      'su_local_foot_action[0][title]' => 'Action Link',
+      'su_local_foot_social[0][uri]' => 'http://facebook.com',
+      'su_local_foot_social[0][title]' => 'Facebook Social Link',
+      'Primary Links Header' => 'Primary links header',
+      'su_local_foot_primary[0][uri]' => 'http://google.com',
+      'su_local_foot_primary[0][title]' => 'Primary Link',
+      'Secondary Links Header' => 'Secondary Links Header',
+      'su_local_foot_second[0][uri]' => 'http://google.com',
+      'su_local_foot_second[0][title]' => 'Secondary Link',
+      'su_local_foot_f_intro[0][value]' => '<p>Lorem Ipsum</p>',
+      'Form Action URL' => 'http://google.com',
+      'Signup Button Text' => 'Sign Me Up',
+    ];
+
+    foreach ($fields as $selector => $value) {
+      $I->fillField($selector, $value);
+    }
+
+    $I->click('Save');
+    $I->runDrush('cache-rebuild');
```

Comment:
I suppose since we have access to the Drupal API we should use `drupal_flush_all_caches()` or the equivalent function instead of drush calls for this kind of thing.

Reply:
I like doing the drush call personally because it simulates a person running it. but its all the same in the end. :man_shrugging: